src/brogue/Recordings.c

```diff
@@ -217,34 +222,27 @@ static void writeHeaderInfo(char *path) {
 }
 
 void flushBufferToFile() {
-    if (currentFilePath[0] == '\0') {
-        return;
-    }
+    if (!rogue.playbackMode && !currentFilePath[0] == '\0') {
+        short i;
+        FILE *recordFile;
 
-    short i;
-    FILE *recordFile;
+        lengthOfPlaybackFile += locationInRecordingBuffer;
+        writeHeaderInfo(currentFilePath);
 
-    if (rogue.playbackMode) {
-        return;
-    }
+        if (locationInRecordingBuffer != 0) {
 
-    lengthOfPlaybackFile += locationInRecordingBuffer;
-    writeHeaderInfo(currentFilePath);
+            recordFile = fopen(currentFilePath, "ab");
 
-    if (locationInRecordingBuffer != 0) {
-
-        recordFile = fopen(currentFilePath, "ab");
-
-        for (i=0; i<locationInRecordingBuffer; i++) {
-            putc(inputRecordBuffer[i], recordFile);
-        }
+            for (i=0; i<locationInRecordingBuffer; i++) {
+                putc(inputRecordBuffer[i], recordFile);
+            }
 
-        if (recordFile) {
-            fclose(recordFile);
+            if (recordFile) {
+                fclose(recordFile);
+            }
         }
-
-        locationInRecordingBuffer = 0;
     }
+    locationInRecordingBuffer = 0;
```

Comment:
Don't understand the rest of the fix, isn't the critical part just moving this line to the top of the function?

Reply:
Yeah, you just need to run this regardless of if clauses.
However, I think `if (locationInRecordingBuffer < INPUT_RECORD_BUFFER_MAX_SIZE) {` is also worthwhile to guard against any future mistakes causing an overrun.